cpp-client/deephaven/dhcore/include/public/deephaven/dhcore/container/container.h

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2016-2025 Deephaven Data Labs and Patent Pending
+ */
+
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <iostream>
+#include <string>
+#include <utility>
+#include "deephaven/dhcore/types.h"
+#include "deephaven/dhcore/utility/utility.h"
+
+namespace deephaven::dhcore::container {
+/**
+ * Forward declaration
+ * @tparam T
+ */
+template<typename T>
+class Container;
+
+class ContainerVisitor {
+protected:
+  using DateTime = deephaven::dhcore::DateTime;
+  using LocalDate = deephaven::dhcore::LocalDate;
+  using LocalTime = deephaven::dhcore::LocalTime;
+
+public:
+  virtual ~ContainerVisitor() = default;
+  virtual void Visit(const Container<char16_t> *) = 0;
+  virtual void Visit(const Container<int8_t> *) = 0;
+  virtual void Visit(const Container<int16_t> *) = 0;
+  virtual void Visit(const Container<int32_t> *) = 0;
+  virtual void Visit(const Container<int64_t> *) = 0;
+  virtual void Visit(const Container<float> *) = 0;
+  virtual void Visit(const Container<double> *) = 0;
+  virtual void Visit(const Container<bool> *) = 0;
+  virtual void Visit(const Container<std::string> *) = 0;
+  virtual void Visit(const Container<DateTime> *) = 0;
+  virtual void Visit(const Container<LocalDate> *) = 0;
+  virtual void Visit(const Container<LocalTime> *) = 0;
+};
+
+class ContainerBase : public std::enable_shared_from_this<ContainerBase> {
+public:
+  explicit ContainerBase(size_t size) : size_(size) {}
```

Comment:
I was going back to front and made some comments earlier about arguments in inline definitions in header files where arguments can be made const; not sure what we are doing in general or what we said about our coding convention, but seems like we consistently are not doing that for headers.  Obviously in a declaration it doesn't make a difference, but these are inline definitions so just a note to discuss about that.  I am not strongly against doing what you are doing here for such small methods, meaning the value of adding const is not that much.  Is more like it attracts my attention during review so need to either adjust or maybe discuss

Reply:
we discussed over slack